Updating docs with notes about documentation. (#203)
junit-pioneer/junit-pioneer: CONTRIBUTING.md

### Branching Strategy

By default, development happens in branches, which are merged via pull requests.
Special cases, like fixing problems with the CI pipeline, are of course exempt from this guideline.

Please make sure to give branches a meaningful name!
As an example, the one creating this documentation was called `branching-merging-documentation`.

### Documentation

When a new extension was added (or significant changes were made) please update the documentation.

> To add the documentation for a new extension add a new entry in the `docs/docs-nav.yml` file and add a new `.adoc` file for the new extension.

Reviewer: i think we can remove all the `new` in this line. because it also applies to existing extensions. and adding an entry implies for me that it is somehwat new :)

Reviewer: ```suggestion
To add documentation for an undocumented extension add an entry in the `docs/docs-nav.yml` file and create a new `.adoc` file for the extension.
```

To ensure that a linebreak is properly displayed in the resulting documentation ensure to write **one and only one sentence per line**.

> Do this also for very short sentences like _"Tests must be fast!"_.

Reviewer: ```suggestion
This rule also applies for very short sentences like _"Tests must be fast!"_.
```

> As long as there is no empty line is between to lines of text then the sentences will be displayed like they were written in one line, but will be wraped dynamically of the readers screen resolution.

Reviewer: ```suggestion
If long as text blocks are not separated by an empty line, they are treated as one paragraph and without line breaks, regardless existing ones. This means you need to apply at least an empty line if you want to have text separated.

For further information regarding formatting, and how to use AsciiDoctor please use the [Writers Guide](https://asciidoctor.org/docs/asciidoc-writers-guide/).
```
